box_error.go

```diff
@@ -0,0 +1,274 @@
+package tarantool
+
+import (
+	"fmt"
+)
+
+const errorExtID = 3
+
+// BoxError is a type representing Tarantool `box.error` object: a single
+// MP_ERROR_STACK object with a link to the previous stack error.
+// See https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_error/error/
+//
+// Since 1.10.0
+type BoxError struct {
+	// Type is error type that implies its source (for example, "ClientError").
+	Type string
+	// File is a source code file where the error was caught.
+	File string
+	// Line is a number of line in the source code file where the error was caught.
+	Line uint32
+	// Msg is the text of reason.
+	Msg string
+	// Errno is the ordinal number of the error.
+	Errno uint32
+	// Code is the number of the error as defined in `errcode.h`.
+	Code uint32
+	// Fields are additional fields depending on error type. For example, if
+	// type is "AccessDeniedError", then it will include "object_type",
+	// "object_name", "access_type".
+	Fields map[interface{}]interface{}
+	// Prev is the previous error in stack.
+	Prev *BoxError
+}
+
+// Error converts a BoxError to a string.
+func (boxError *BoxError) Error() string {
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#receiver-names

```suggestion
func (e *BoxError) Error() string {
```

Reply:
Changed